### PR TITLE
Support custom configuration outside of GitHub

### PR DESCRIPTION
**Pull request by salbertson**
This could be used to centralize style guides.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,8 +14,8 @@ gem "coffeelint"
 gem "email_validator"
 gem "faraday"
 gem "font-awesome-rails"
-gem "haml_lint"
 gem "haml-rails"
+gem "haml_lint"
```
Comment (gylaz) [inline Gemfile]: This change seems unrelated?

Comment (salbertson) [inline Gemfile]: Yes, just sorted the gems.

File: app/models/repo_config.rb [hunk 2]
```diff
@@ -134,6 +134,14 @@ def load_file(file_path, file_type)
     end
   end
 
+  def get_file_content(file_path_or_url)
+    if file_path_or_url.match("^http")
```
Comment (gylaz) [inline app/models/repo_config.rb]: What do you think about using `URI::regexp(%w(http https)).match(file_path_or_url )` to check for url?

**Comment by gylaz** [comment]
LGTM

**Comment by teoljungberg** [comment]
I'm super into this idea!

File: app/models/repo_config.rb [hunk 3]
```diff
@@ -134,6 +134,14 @@ def load_file(file_path, file_type)
...
+      Faraday.new.get(file_path_or_url).body
```
Comment (gylaz) [inline app/models/repo_config.rb]: As discussed IRL, this would only work for public repos/gists. What if we supported private repos too? We could use Octokit library to fetch those files.

This way users could paste the link they grab from GitHub:
```yaml
ruby:
  config_file: https://github.com/thoughtbot/hound/blob/master/.ruby-style.yml
```
Thoughts?

Comment (teoljungberg) [inline app/models/repo_config.rb]: Also, what happens if the URL is a 404, do we want to guard against that somehow? I would like it to follow the same scheme as it does now; to mark the build as a failure with an error message saying that the provided config is invalid.

Comment (teoljungberg) [inline app/models/repo_config.rb]: I'm into the idea @gylaz suggested, that allows for private style guides which I can see being attractive to some

Comment (salbertson) [inline app/models/repo_config.rb]: I like the idea. It will take some reworking of `RepoConfig` so I'll try it in a different branch.

Comment (teoljungberg) [inline app/models/repo_config.rb]: What do you think about waiting on doing that until #950 has landed?

---
Teo Ljungberg

File: spec/models/repo_config_spec.rb [hunk 4]
```diff
@@ -196,21 +196,75 @@
...
+          )
+
+          result = repo_config.for("ruby")
+
+          expect(result).to eq("LineLength" => { "Max" => 90 })
+        end
+      end
+
+      context "when config is an invalid URL" do
+        it "returns parsed config at URL" do
```
Comment (teoljungberg) [inline spec/models/repo_config_spec.rb]: Should this be `it "raises an error"`?

Comment (salbertson) [inline spec/models/repo_config_spec.rb]: Yes, thank you.
